### PR TITLE
Upgrade to Rails 5.1

### PR DESCRIPTION
**Pull request by tjwp**
This upgrade is broken into several commits:
1. update gems for Rails 5.1 -- and bundler 1.16
1. run `rails app:update:bin` and edit results
  - the approach for setting the default port had to change
  - the `to_prepare` callback had to be replaced (previously deprecated)
1. run `rails app:update:config` and edit results

Prime: @jturkel

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -1,13 +1,6 @@
...
-# Pick the frameworks you want:
-require 'active_model/railtie'
-require 'active_job/railtie'
-require 'active_record/railtie'
-require 'action_controller/railtie'
-require 'action_mailer/railtie'
-require 'action_view/railtie'
+require_relative 'boot'
+
+require 'rails/all'
```
Comment (jturkel) [inline config/application.rb]: Should we go back to the explicit enumeration of requires and leave off action_mailer + active_job + action_cable?

File: bin/yarn [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env ruby
```
Comment (jturkel) [inline bin/yarn]: Do we need this since we're not using yarn?

File: config/cable.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+development:
```
Comment (jturkel) [inline config/cable.yml]: Do we need this since we're not using action cable?

File: bin/yarn (hunk 2 above)
Comment (tjwp) [reply bin/yarn]: Probably not, I'm happy to drop it.

File: config/cable.yml (hunk 3 above)
Comment (tjwp) [reply config/cable.yml]: Probably not, I'm happy to drop it.

File: config/application.rb (hunk 1 above)
Comment (tjwp) [reply config/application.rb]: I didn't think that through. I'll revert this and clean it up.

**Pull request merged**